src/components/modals/ManageProjectAddresses/ManageProjectAddressesModal.tsx

```diff
@@ -16,68 +17,56 @@ import config from '@/configuration';
 import { NetworkWalletAddress } from './NetworkWalletAddress';
 import { AddNewAddress } from './AddNewAddress';
 import { getChainName } from '@/lib/network';
+import { NonEVMChain } from '@/types/config';
 import type { IModal } from '@/types/common';
 
-const networksConfig = config.EVM_NETWORKS_CONFIG;
-const networkIds = Object.keys(networksConfig).map(Number);
+const { CHAINS } = config;
 
-interface IManageProjectAddressesModal extends IModal {
+interface IModalProps extends IModal {
 	project: IProject;
 	setProjects: Dispatch<SetStateAction<IProject[]>>;
 }
 
-export const ManageProjectAddressesModal: FC<IManageProjectAddressesModal> = ({
-	project,
-	setShowModal,
-	setProjects,
-}) => {
-	const [selectedWallet, setSelectedWallet] = useState<IWalletAddress>();
-	const [addresses, setAddresses] = useState<IWalletAddress[]>([]);
+export const ManageProjectAddressesModal: FC<IModalProps> = props => {
+	const { project, setShowModal, setProjects } = props;
+	const [selectedChain, setSelectedChain] = useState<Chain | NonEVMChain>();
+	const [addresses, setAddresses] = useState<IWalletAddress[]>(
+		project.addresses || [],
+	);
 	const { isAnimating, closeModal } = useModalAnimation(setShowModal);
 	const { formatMessage } = useIntl();
-
-	useEffect(() => {
-		let WalletAddr: { [key: number]: IWalletAddress } = {};
-		networkIds.forEach(networkId => {
-			WalletAddr[networkId] = { networkId };
-		});
-		const { addresses } = project;
-		if (!addresses) return;
-		for (let i = 0; i < addresses.length; i++) {
-			const address = addresses[i];
-			if (address.networkId) {
-				WalletAddr[address.networkId] = address;
-			}
-		}
-		setAddresses(Object.values(WalletAddr));
-	}, [project]);
+	const selectedChainType =
+		selectedChain && 'chainType' in selectedChain
+			? selectedChain.chainType
+			: undefined;
 
 	return (
 		<Modal
 			headerIcon={<IconWalletOutline32 />}
 			headerTitle={formatMessage({
-				id: selectedWallet
+				id: selectedChain
 					? 'label.add_new_address'
 					: 'label.manage_addresses',
 			})}
 			closeModal={closeModal}
 			isAnimating={isAnimating}
 			headerTitlePosition='left'
 			backButtonCallback={
-				selectedWallet ? () => setSelectedWallet(undefined) : undefined
+				selectedChain ? () => setSelectedChain(undefined) : undefined
 			}
 		>
 			<ModalContainer>
 				<Content>
-					{selectedWallet ? (
+					{selectedChain ? (
 						<SublineBold>
 							{formatMessage(
 								{
 									id: 'label.chain_address',
 								},
 								{
 									chainName: getChainName(
```

Comment:
Likewise

Reply:
@RamRamez to use useGeneralWallet here as well

Reply:
@aminlatifi We can't use "useGeneralWallet" since it's returning user wallet current chain. In ManageProjectAddressesModal, we are not going to show current chain that user is on, the aim is to set a wallet address for the chain that user has selected.
Here, I used "useGeneralWallet", then I've chosen to set an address for Solana, but as you see, it says Gnosis, because I'm on Gnosis network:
![image](https://github.com/Giveth/giveth-dapps-v2/assets/73441857/325c3113-4067-4d7d-882b-c4bbd90f5cfa)

cc @jainkrati